spec/requests/cart_item_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'CartItem', type: :request do
+  let!(:user) { create(:user, shopping_cart: create(:shopping_cart)) }
+  let!(:product) { create(:product) }
+
+  describe 'POST /cart' do
+    context 'when user signed in' do
+      before do
+        sign_in user
+      end
+
+      it 'has success http status' do
+        post "/cart/#{product.id}"
+        follow_redirect!
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'renders success notification after adding product to cart' do
+        post "/cart/#{product.id}"
+        follow_redirect!
+        expect(response.body).to include("Product #{product.name} has been added to your shopping cart")
+      end
+
+      it 'can add product to shopping cart with specified quantity' do
+        post "/cart/#{product.id}", params: { quantity: 5 }
+        follow_redirect!
+        expect(user.shopping_cart.cart_items.find_by(product_id: product.id).quantity).to eq(5)
+      end
+
+      it 'increase quantity of cart item if same product is added to shopping cart' do
+        2.times do
+          post "/cart/#{product.id}"
+          follow_redirect!
+        end
+        expect(user.shopping_cart.cart_items.find_by(product_id: product.id).quantity).to eq(2)
+      end
+
+      it 'increase quantity of cart item by given number if product is already in cart' do
+        2.times do
+          post "/cart/#{product.id}"
+        end
+        post "/cart/#{product.id}", params: { quantity: 5 }
+        follow_redirect!
+        expect(user.shopping_cart.cart_items.find_by(product_id: product.id).quantity).to eq(7)
+      end
+
+      it 'not allows to provide not numerical value' do
+        post "/cart/#{product.id}", params: { quantity: 'a' }
+        follow_redirect!
+        expect(response.body).to include('Quantity is not a number')
+      end
+
+      it 'convert quantity nil value to quantity equals one' do
+        post "/cart/#{product.id}", params: { quantity: nil }
```

Comment:
dlaczego tego nie wyciagnac do subjecta? ;)

Reply:
Okej, udało mi się powalczyć z tym subjectem. :) Zdecydowanie lepiej to wygląda

Reply:
widze jeszcze kilka miejsc, gdzie brakuje subjecta. Nie wrzucaj go w before. W koncu bedzie przypadek kiedy trzeba bedzie przed requestem coś zamockować i pojawi się problem - trzeba bedzie specke refactorowac :). **Wywolanie subjecta powinno byc explicit!**

Reply:
Zmieniłem wywołanie na explicit, będę o tym pamiętać :) Dodałem też subjecty do model i system specek. Rozumiem że tam ich brakowało?